backend/app/services/posting_service.rb

```diff
@@ -65,14 +65,71 @@ def prefill(user:)
         applicant = Applicant.find_by(utorid: utorid)
         return { utorid: utorid } unless applicant
 
-        {
+        # This is the basic data we can get if the applicant already exists
+        data = {
             utorid: utorid,
             student_number: applicant.student_number,
             first_name: applicant.first_name,
             last_name: applicant.last_name,
             email: applicant.email,
             phone: applicant.phone
         }
+
+        existing_application = Application.find_by(posting: @posting)
+        if existing_application
+            application_service =
+                ApplicationService.new application: existing_application
+            data.merge! application_service.prefilled_data
```

Comment:
Do I understand correctly that we are 

1) Creating a new application by providing existing application (from our search by posting) as a parameter to the `ApplicationService`'s `new` method,
2) then merging the prefilled data of newly created application with the `data` described above (that we got from the applicant data)
3) And that happens only if such an application existed for a posting?

And the reason for doing that is so we could extract some posting-specific data and applicant-related data? How did it work before? Was it just prefilling the utorid?

Reply:
We aren't actually creating an a new application here. We're looking 1) at a current application, and if there is none 2) at a past application. Then we extract relevant information from those applications.